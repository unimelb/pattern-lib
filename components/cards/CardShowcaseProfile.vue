<template>
  <div class="card card--bdr card--showcase-profile">
    <div
      v-show="thumb && thumb.length > 0 && isHidden"
      :style="{ backgroundImage: `url(${thumb})` }"
      class="card__thumb"
      aria-label="Profile Image"
    />
    <div
      :class="classes"
      class="card__container"
      @click="changeIcon">
      <div
        v-if="!isHidden && thumb"
        :style="{ backgroundImage: `url(${thumb})` }"
        class="card__thumb--inner"
        aria-label="Profile Image"
      />
      <div class="card__inner">
        <div class="card__titles">
          <h6 class="card__title">{{ title }}</h6>
          <p class="card__sub-title">{{ subTitle }}</p>
        </div>
        <div>
          <SvgIcon
            v-if="bio"
            :name="iconName"
            aria-label="chevron"
            width="14px"
            height="14px"
          />
        </div>
      </div>
    </div>

    <transition name="slide-fade">
      <div
        v-show="!isHidden"
        class="card__bio">
        <VideoEmbed
          v-if="video"
          :src="video"/>
        <p class="card__bio--text">{{ bio }}</p>
      </div>
    </transition>
  </div>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';
import VideoEmbed from '../embed/VideoEmbed.vue';
import { IMAGE_PLACEHOLDER_SHORT } from '../../utils/placeholders';

export default {
  name: 'Showcase',
  components: { SvgIcon, VideoEmbed },
  props: {
    thumb: {
      type: String,
      default: IMAGE_PLACEHOLDER_SHORT,
    },
    title: {
      type: String,
      default: '',
    },
    subTitle: {
      type: String,
      default: '',
    },
    video: {
      type: String,
      default: '',
    },
    bio: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      isHidden: true,
      iconName: 'chevron-up',
      isFixed: false,
    };
  },
  computed: {
    classes() {
      return {
        'card__container--fixed': this.isFixed,
      };
    },
  },
  methods: {
    changeIcon() {
      if (this.bio !== '') {
        this.isHidden = !this.isHidden;
        this.isFixed = !this.isFixed;
        if (this.iconName === 'chevron-up') {
          this.setIconDown();
        } else {
          this.setIconUp();
        }
      }
    },
    setIconUp() {
      this.iconName = 'chevron-up';
    },
    setIconDown() {
      this.iconName = 'chevron-down';
    },
  },
};
</script>
