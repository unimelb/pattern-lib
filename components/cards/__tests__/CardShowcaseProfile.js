import {
  shallow,
} from 'vue-test-utils';
import {
  toHaveNoViolations,
} from 'jest-axe';
import CardShowcaseProfile from '../CardShowcaseProfile.vue';

expect.extend(toHaveNoViolations);

describe('CardShowcaseProfile', () => {
  it('should match snapshot', () => {
    const result = shallow(CardShowcaseProfile).element;
    expect(result).toMatchSnapshot();
  });

  it('should have default props and correct types thumb/title/subTitle/bio', () => {
    const wrapper = shallow(CardShowcaseProfile);
    const {
      thumb,
      title,
      subTitle,
      bio,
    } = wrapper.vm.$options.props;

    expect(thumb.type).toEqual(String);
    expect(wrapper.props().thumb).toBe('https://via.placeholder.com/800x630');

    expect(title.type).toBe(String);
    expect(wrapper.props().title).toBe('');

    expect(subTitle.type).toBe(String);
    expect(wrapper.props().subTitle).toBe('');

    expect(bio.type).toBe(String);
    expect(wrapper.props().bio).toBe('');
  });

  it('should render thumb from prop with correct type', () => {
    const thumb = 'http://';
    const wrapper = shallow(CardShowcaseProfile, {
      propsData: {
        thumb: 'http://',
      },
    });

    expect(typeof wrapper.props().thumb).toBe('string');
    expect(wrapper.props().thumb).toBe(thumb);
    expect(wrapper.find('img').html()).toContain(thumb);
  });

  it('should render title from prop with correct type', () => {
    const title = 'Test title';
    const wrapper = shallow(CardShowcaseProfile, {
      propsData: {
        title: 'Test title',
      },
    });

    expect(typeof wrapper.props().title).toBe('string');
    expect(wrapper.props().title).toBe(title);
    expect(wrapper.find('h6').text()).toBe(title);
  });

  it('should render subTitle from prop with correct type', () => {
    const subTitle = 'Test subTitle';
    const wrapper = shallow(CardShowcaseProfile, {
      propsData: {
        subTitle: 'Test subTitle',
      },
    });

    expect(typeof wrapper.props().subTitle).toBe('string');
    expect(wrapper.props().subTitle).toBe(subTitle);
    expect(wrapper.find('.card__sub-title').text()).toBe(subTitle);
  });

  it('should render bio prop with correct type', () => {
    const bio = 'Test bio';
    const wrapper = shallow(CardShowcaseProfile, {
      propsData: {
        bio: 'Test bio',
      },
    });

    expect(typeof wrapper.props().bio).toBe('string');
    expect(wrapper.props().bio).toBe(bio);
    expect(wrapper.find('.card__bio').text()).toBe(bio);
  });
});
