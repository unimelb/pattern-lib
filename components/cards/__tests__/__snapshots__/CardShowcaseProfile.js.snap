// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardShowcaseProfile should match snapshot 1`] = `
<div
  class="card card--bdr card--showcase-profile"
>
  <div
    aria-label="Profile Image"
    class="card__thumb"
    style="background-image: url(https://matrix-cms.unimelb.edu.au/__data/assets/image/0028/85159/staff-placeholder-short.png);"
  />
   
  <div
    class="card__container"
  >
    <!---->
     
    <div
      class="card__inner"
    >
      <div
        class="card__titles"
      >
        <h6
          class="card__title"
        >
          
        </h6>
         
        <p
          class="card__sub-title"
        >
          
        </p>
      </div>
       
      <div>
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="card__bio"
    name="slide-fade"
    style="display: none;"
  >
    <!---->
     
    <p
      class="card__bio--text"
    >
      
    </p>
  </div>
</div>
`;
